dagmc_stats/DagmcFile.py

```diff
@@ -106,6 +109,34 @@ def __set_entityset_ranges(self):
                 self._my_moab_core.get_entities_by_type_and_tag(self.root_set, types.MBENTITYSET,
                                                                 self.dagmc_tags['geom_dim'], [dimension])
 
+    def get_meshset_by_id(self, dim, ids=[]):
+        """Get meshset of the geometry with specified dimension and ids
+
+        inputs
+        ------
+        dim : dimension of the meshset
```

Comment:
should specify if this is an integer or string and what the valid options are for `dim`.

Reply:
Right now it's set up to only accept `Volumes` or `Surfaces` or `Curves` or `Nodes` (case sensitive and plural). I can see a user trying to use something like `volume` or an integer (0-3), which would cause an error because you can't call those keys in the `entityset_ranges` dictionary on line 132. One suggestion to get around this is to make a few changes:

1. change `self.entityset_types` to use all lower case strings ('nodes', 'curves', 'surfaces', 'volumes'} so that you can convert any string passed into `dim` to all lowercase if it's a string: `dim.lower()`
2. in this method, the first check should be whether `dim` is an integer or a string. If it's an integer (`[0, 1, 2, 3]`) then just call the correct string from `entityset_types` to use the correct string in the rest of the method. If it's a string, convert it to `lower()` and make sure it's a valid string according to the dictionary keys. You'll probably also want a check to all both the plural or singular form of the words (eq, both `volume` and `volumes`).
3. If neither the integer nor the string for `dim` is valid, we should add a warning or error.